neuronsimulator/forms.py

```diff
@@ -1,8 +1,97 @@
+    resp_neu_ids = forms.CharField(
+        max_length=3000, required=False, widget=forms.HiddenInput()
+    )
+    nt = forms.IntegerField(
+        initial=1000,
+        label="Number of time points(nt)",
+        validators=[MinValueValidator(1)],
+    )
+    # use t_max to replace dt
+    t_max = forms.DecimalField(
+        initial=100,
+        label="Maximum time(t_max)",
+        validators=[MinValueValidator(0)],
+        help_text="value requirement: t_max>dur",
+    )
+    stim_type = forms.ChoiceField(
+        choices=STIM_TYPE_CHOICES,
+        widget=forms.Select,
+        label="Type of standard stimulus(stim_type)",
+        initial="rectangular",
+    )
+    dur = forms.DecimalField(
+        initial=2.0,
+        label="Duration of the stimulus(dur)",
+        validators=[MinValueValidator(0.1)],
+    )
+
+    def clean(self):
+        cleaned_data = super().clean()
+
+        stim_neu_id = cleaned_data.get("stim_neu_id")
+        resp_neu_ids = cleaned_data.get("resp_neu_ids")
+
+        # a neuron cannot be selected for both stim_neu_id and resp_neu_ids
+        if resp_neu_ids is not None:
+            resp_neu_arr = cleaned_data.get("resp_neu_ids").split(",")
+            if stim_neu_id in resp_neu_arr:
+                raise forms.ValidationError(
+                    "Response neuron(s) included stimulated neuron, please re-select!"
+                )
+
+        # t_max should be larger than dur
+        t_max = float(cleaned_data.get("t_max"))
+        dur = float(cleaned_data.get("dur"))
+        if t_max < dur:
+            raise forms.ValidationError(
+                "t_max value need to be larger than dur, please re-select!"
+            )
+
+    def clean_stim_neu_id(self):
+        neurons = Neuron.objects.values_list("name", flat=True).all()
+        stim_neu_id = self.cleaned_data.get("stim_neu_id", None)
+        if stim_neu_id not in neurons:
+            raise forms.ValidationError("stimulated neuron is not a valid neuron")
+        return stim_neu_id
+
+    def clean_resp_neu_ids(self):
+        neurons = Neuron.objects.values_list("name", flat=True).all()
+        resp_neu_ids = self.cleaned_data.get("resp_neu_ids", None)
+
+        """
+        resp_neu_ids could be None based on wormfunconn source code
+        but require at least one resp neuron at this stage of test
+        will deal with this case later: resp_neu_ids=None and use a threshold value
+        """
+        if resp_neu_ids is None:
+            raise forms.ValidationError(
+                "need to select at least one response neuron (for testing!)"
+            )
+        else:
+            resp_neu_arr = resp_neu_ids.split(",")
+            for resp_neuron in resp_neu_arr:
+                if resp_neuron not in neurons:
+                    raise forms.ValidationError(
+                        "found invalid neuron in response neuron list, please re-select!"
+                    )
```

Comment:
NB. Given multiple possibilities, I recommend collection all the errors first and report which ones are invalid.

```
            inv_neurs = []
            for resp_neuron in resp_neu_arr:
                if resp_neuron not in neurons:
                    inv_neurs.append(resp_neuron)
            if len(inv_neurs) > 0:
                raise forms.ValidationError(
                    f"Invalid response neuron(s) encountered: [{', '.join(inv_neurs)}]"
                )
```